tilequeue/rawr.py

```diff
@@ -580,22 +580,26 @@ class RawrS3Sink(object):
 
     """Rawr sink to write to s3"""
 
-    def __init__(self, s3_client, bucket, prefix, suffix):
+    def __init__(self, s3_client, bucket, prefix, suffix, metadata=None):
         self.s3_client = s3_client
         self.bucket = bucket
         self.prefix = prefix
         self.suffix = suffix
+        self.metadata = metadata
 
     def __call__(self, rawr_tile):
         payload = make_rawr_zip_payload(rawr_tile)
         location = make_rawr_s3_path(rawr_tile.tile, self.prefix, self.suffix)
-        self.s3_client.put_object(
-                Body=payload,
-                Bucket=self.bucket,
-                ContentType='application/zip',
-                ContentLength=len(payload),
-                Key=location,
+        put_opts = dict(
+            Body=payload,
+            Bucket=self.bucket,
+            ContentType='application/zip',
+            ContentLength=len(payload),
+            Key=location,
         )
+        if self.metadata:
+            put_opts['Metadata'] = self.metadata
```

Comment:
S3's `Metadata` is different from the `Tagging`. Check out their [docs here](https://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.put_object). You want to do something like this instead:

```python
from urllib.parse import urlencode
if self.metadata:
    put_opts['Tagging'] = urlencode(self.metadata)
```

(note that boto3 `put_object()` docs say that "The tag-set must be encoded as URL Query parameters")

Reply:
5ed0b16